src/main/java/net/joshka/junit/json/params/JsonArgumentsProvider.java

```diff
@@ -38,16 +42,24 @@ public Stream<? extends Arguments> provideArguments(ExtensionContext extensionCo
 
     private Stream<? extends Arguments> getArguments(String value) throws IOException {
         try (Reader reader = new StringReader(value)) {
-            return values(reader).map(Arguments::of);
+            if (this.targetClass.equals(Object.class)) {
+                return values(reader, this.isArrayClass)
+                        .map(Arguments::of);
+            }
+            return values(reader, this.isArrayClass)
+                    .map(json -> JsonConverter.convert(json, this.targetClass))
+                    .map(Arguments::of);
         }
     }
 
-    private static Stream<JsonValue> values(Reader reader) {
+    private static Stream<JsonValue> values(Reader reader, boolean arrayClass) {
```

Comment:
Use `this.isArrayClass` directly in this method rather than calling a private method with value which is stored in the class.